x-pack/plugins/aiops/public/get_document_stats.ts

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { each, get } from 'lodash';
+import type * as estypes from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+import { isPopulatedObject } from '@kbn/ml-is-populated-object';
+import { buildBaseFilterCriteria } from './query_utils';
+
+export interface DocumentCountStats {
+  interval?: number;
+  buckets?: { [key: string]: number };
+  timeRangeEarliest?: number;
+  timeRangeLatest?: number;
+  totalCount: number;
+}
+
+export interface OverallStatsSearchStrategyParams {
+  earliest?: number;
+  latest?: number;
+  intervalMs?: number;
+  index: string;
+  timeFieldName?: string;
+  runtimeFieldMap?: estypes.MappingRuntimeFields;
+  fieldsToFetch?: string[];
+}
+
+export const getDocumentCountStatsRequest = (params: OverallStatsSearchStrategyParams) => {
+  const {
+    index,
+    timeFieldName,
+    earliest: earliestMs,
+    latest: latestMs,
+    runtimeFieldMap,
+    // searchQuery,
+    intervalMs,
+    fieldsToFetch,
+  } = params;
+
+  const size = 0;
+  const filterCriteria = buildBaseFilterCriteria(timeFieldName, earliestMs, latestMs, {
+    match_all: {},
```

Comment:
Should this be using `searchQuery` instead of `match_all`? If not, would be good to clean up searchQuery or add a comment.

Reply:
Yep - I'll be adding search support in a follow up 👍